Add support for M3U playlists (#182)
geigerzaehler/beets-alternatives: README.md

Now, if you move the `/music/` folder to another location, the links
will continue working

### Playlists

You can specify playlists in each collection so that they are copied over
and automatically redirected to tracks inside the collection.
Any tracks in the playlist that are not in the collection are omitted.

```yaml
alternatives:
  myplayer:
    directory: /player
    paths:
      default: $album/$title
    formats: aac mp3
    query: "onplayer:true"
    playlist_dir: /player/playlists

>     playlists:

Reviewer: The setting names sound ambiguous to me. And how using different term and nesting them?
```yaml
playlists:
  sources:
    - ...
   dest: ./playlists
```

Author: Ah, good catch! I'll rename them to `playlist_dir` -> `playlist_dest_dir` and `playlists` -> `playlist_sources`

and automatically redirected to tracks inside the collection.
Any tracks in the playlist that are not in the collection are omitted.

```yaml
alternatives:
  myplayer:
    directory: /player
    paths:
      default: $album/$title
    formats: aac mp3
    query: "onplayer:true"
    playlist_dest_dir: /player/playlists
    playlist_sources:
      - /path/to/playlist.m3u
      - /path/to/another-playlist.m3u8
      - /path/to/directory/containing/playlists/*.m3u
      - /path/to/directory/containing/directories/of/playlists/**/*.m3u
    removable: true
```

Say you had a playlist that looked like this:

```m3u
#EXTM3U
/home/user/Music/Michael Jackson/Thriller/Beat It.flac
/home/user/Music/Michael Jackson/Thriller/Billie Jean.flac
/home/user/Music/Michael Jackson/Thriller/Human Nature.flac
```

Using this config, the `beet alt update myplayer` command would then produce
this playlist file under `/player/playlists`:
```m3u
#EXTM3U
../Thriller/Beat It.mp3
../Thriller/Billie Jean.mp3
../Thriller/Human Nature.mp3
```

You can specify the `playlist_path_type` configuration option to be `absolute`
if you need your playlist file to use absolute paths.

> Currently, playlists in alternative collections will always be replaced
> with playlists defined from the config.
> However, there is a plan to allow collection playlists to be merged with
> main playlists.

Reviewer: It took me some time to understand what this meant. Let's clarify this a bit

```suggestion
On every `beet alt update`, the existing playlists in
`playlist_dest_dir` are deleted before the are regenerated from
`playlist_sources`.
```

  ImageMagick estimates input quality (using 92 if unknown), PIL uses 75.
  Default: 0 (disabled) (optional)

* **`playlist_dir`** The root directory to store playlists under.
  Relative paths are resolved with respect to the collection's root directory.
  If omitted it defaults to a directory named `playlists` under the

>   collection directory. (optional)

Reviewer: We should mention that the directory contents is removed before syncing so that old playlists are not retained.

Author: Agreed - I'll add a warning here

* **`playlists`** A list of paths to m3u, m3u8, or directories containing such

>   files. These playlists are copied into a directory specified by `playlist_dir`,

Reviewer: How about just supporting globs (using [`Path.glob`](https://docs.python.org/3.14/library/pathlib.html#pattern-language))? This would simplify the usage as well as the implementation.

  ImageMagick estimates input quality (using 92 if unknown), PIL uses 75.
  Default: 0 (disabled) (optional)

* **`playlist_dest_dir`** The root directory to store playlists under.
  Any playlists found in this directory **will be deleted** before repopulating them.
  Relative paths are resolved with respect to the collection's root directory.
  If omitted it defaults to a directory named `playlists` under the
  collection directory. (optional)

> * **`playlist_sources`** A list of paths or glob patterns to m3u, m3u8,
>   or directories containing such files.

Reviewer: ```suggestion
* **`playlist_sources`** A list of paths or glob patterns to m3u and m3u8 playlist files.
```

Directories are not recursively read any more
